web-common/src/features/sources/extract-table-name.spec.ts

```diff
@@ -0,0 +1,76 @@
+import {
+  extractFileExtension,
+  extractTableName,
+  sanitizeEntityName,
+} from "@rilldata/web-common/features/sources/extract-table-name";
+import { describe, it, expect } from "vitest";
+
+describe("extract-table-name", () => {
+  describe("should extract and sanitise table name", () => {
+    for (const variation of Variations) {
+      it(`${variation.title} path=${variation.path}`, () => {
+        expect(sanitizeEntityName(extractTableName(variation.path))).toBe(
+          variation.expectedFileName
+        );
+      });
+    }
+  });
+
+  describe("should extract extension", () => {
+    for (const variation of Variations) {
+      it(`${variation.title} path=${variation.path}`, () => {
+        expect(extractFileExtension(variation.path)).toBe(
+          variation.expectedExtension
+        );
+      });
+    }
+  });
+});
+
+function getVariations(
+  fileName,
+  expectedFileName,
+  expectedExtension = ".parquet"
```

Comment:
It'd help readability of the code below if you explicitly set a value in all cases and didn't have a default